index.bs

```diff
@@ -439,6 +440,157 @@ spec: webidl
     it MUST remove it.
     </li>
   </ol>
+
+  <section>
+    <h3 id="storage-manipulation">Storage Manipulation Algorithms</h3>
+
+    <p>
+      Other specifications can use the algorithms defined in this section to
+      manipulate the <a>permission store</a> from their algorithms.
+    </p>
+
+    <div class="example" id="example-storage-manipulation">
+      <p>
+        The [[notifications]] API can check for the user having granted
+        permission using:
+      </p>
+      <blockquote>
+        If the {{"notifications"}} <a>permission's state</a> is not
+        {{"granted"}}, reject <var>promise</var> with a {{TypeError}} exception
+        and terminate these substeps.
```

Comment:
Gecko used to have a domain-based permission model as well, even across HTTP/HTTPS. We recently fixed that to make it origin-based.

(My comment about `document.domain` was slightly misleading. With `document.domain` those settings objects can have different origins, period. They would compare as equal for "same origin-domain", but might not always do so, and I believe we typically use "same origin" for permissions though as you noted there is still some variance there.)